Shared/Tests.Sync.Shared/SessionTests.cs

```diff
@@ -117,6 +116,49 @@ public void Session_Error_ShouldPassCorrectSession()
             });
         }
 
+        [Test]
+        public void Session_DivergingHistories_ShouldRaiseClientResetException()
+        {
+            AsyncContext.Run(async () =>
+            {
+                var realm = await SyncTestHelpers.GetFakeRealm(isUserAdmin: true);
+                var tcs = new TaskCompletionSource<Tuple<Session, ClientResetException>>();
+                Session.Error += (sender, e) =>
```

Comment:
What happens if you repeat the unit test, with the error handler being added twice?